client/packages/reports/src/DetailView/DetailView.tsx

```diff
@@ -128,6 +128,7 @@ const DetailViewInner = ({
         isPrinting={isPrinting}
       />
       <ReportArgumentsModal
+        key={report.id}
```

Comment:
don't see why the `key` is necessary, or what value that adds 🤷

Reply:
Sorry, @mark-prins, this was from my PR (#4501 ) which I've merged into here already (I intended to wait, had an absent-minded moment).

Regarding the key:
> add a "key" to the ReportArgsModal so that it'll force a remount (and therefore data reset) when the report type (id) changes

This was because I removed a "clearData" function elsewhere and this was a way to ensure the form gets reset when the report type changes.

Reply:
oh, nice! slick way to get the component to remount